rts/Sim/Misc/DefinitionTag.h

```diff
@@ -174,11 +178,24 @@ template<typename T>
 class DefTagBuilder
 {
 public:
-	DefTagBuilder(DefTagTypedMetaData<T>* d) : data(d) {}
+	DefTagBuilder(DefType* dt, DefTagTypedMetaData<T>* d) : def(dt), data(d) {}
 
 	const DefTagMetaData* GetData() const { return data; }
 	operator T() const { return data->GetData(); }
 
+	// Implementation note: although not strictly required, calling asReference at
+	// the beggining of the fluent chain means that BuiltDefTag only needs a
+	// single constructor which accepts DefTagBuilder by reference (as opposed to
+	// multiple constructors).
```

Comment:
what sort of multiple constructors? it seems to only be called in lines 384 and 402 below, both of which seem to return an xvalue at a glance?

Reply:
I like the idea of the chain being an xvalue because conceptually it already is. Here's a commit https://github.com/sprunk/spring-1/commit/b7d7104e0b4a8e22dfb28f4efdda1c6aed048123

Reply:
Yes. In this case the reference can't be const because we modify `addTagCalled`. Though this seems to be a debug thing (only used in the assert next to it) so another solution would have been to do `mutable addTagCalled` and have a const reference.